src/data/hooks.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  getIndexTimeoff,
+  getShowEmployment,
+  TimeoffStatus,
+} from '@/src/client';
+import { useClient } from '@/src/context';
+import { ContractAmendmentParams } from '@/src/flows/ContractAmendment/types';
+import { Client } from '@hey-api/client-fetch';
+import { useQuery } from '@tanstack/react-query';
+
+type UseEmployment = Pick<ContractAmendmentParams, 'employmentId'>;
+
+/**
+ * Hook to retrieve employment details for a specific employment ID.
+ *
+ * @param {Object} params - The parameters for the query.
+ * @param {string} params.employmentId - The ID of the employment to fetch details for.
+ * @returns {UseQueryResult<any, unknown>} - The result of the query, including the employment details.
+ *
+ * @example
+ * const { data, isLoading, error } = useEmploymentQuery({
+ *   employmentId: '12345',
+ * });
+ *
+ * if (isLoading) {
+ *   return <div>Loading...</div>;
+ * }
+ *
+ * if (error) {
+ *   return <div>Error loading employment details</div>;
+ * }
+ *
+ * return (
+ *   <div>
+ *     <h1>{data.full_name}</h1>
+ *   </div>
+ * );
+ */
+export const useEmploymentQuery = ({ employmentId }: UseEmployment) => {
+  const { client } = useClient();
+  return useQuery({
+    queryKey: ['employment', employmentId],
+    retry: false,
+    queryFn: () => {
+      return getShowEmployment({
+        client: client as Client,
+        headers: {
+          Authorization: ``,
+        },
+        path: { employment_id: employmentId },
+      });
+    },
+    select: ({ data }) => data,
+  });
+};
+
+/**
+ * Hook to retrieve time off data for a specific employment.
+ *
+ * @param {Object} params - The parameters for the query.
+ * @param {string} [params.employmentId] - The ID of the employment to fetch time off data for.
+ * @param {TimeoffStatus} [params.status] - The status of the time off requests to filter by (e.g., 'approved', 'pending').
+ * @returns {UseQueryResult<any, unknown>} - The result of the query, including the time off data.
+ *
+ * @example
+ * const { data, isLoading, error } = useTimeOffQuery({
+ *   employmentId: '12345',
+ *   status: 'approved',
+ * });
+ *
+ * if (isLoading) {
+ *   return <div>Loading...</div>;
+ * }
+ *
+ * if (error) {
+ *   return <div>Error loading time off data</div>;
+ * }
+ *
+ * return (
+ *   <div>
+ *     {data.map((timeOff) => (
+ *       <div key={timeOff.id}>{timeOff.reason}</div>
+ *     ))}
+ *   </div>
+ * );
+ */
```

Comment:
In this particular case, I think the examples, for these functions are too much and don't add anything new IMO, so I would remove it.